src/pages/studyView/resources/FilesAndLinks.tsx

```diff
@@ -0,0 +1,329 @@
+import * as React from 'react';
+import {
+    Column,
+    default as LazyMobXTable,
+} from 'shared/components/lazyMobXTable/LazyMobXTable';
+import { observer } from 'mobx-react';
+import _ from 'lodash';
+import internalClient from 'shared/api/cbioportalInternalClientInstance';
+import { Else, If, Then } from 'react-if';
+import { WindowWidthBox } from '../../../shared/components/WindowWidthBox/WindowWidthBox';
+import LoadingIndicator from 'shared/components/loadingIndicator/LoadingIndicator';
+import {
+    getSampleViewUrlWithPathname,
+    getPatientViewUrlWithPathname,
+} from 'shared/api/urls';
+import { getAllClinicalDataByStudyViewFilter } from '../StudyViewUtils';
+import { StudyViewPageStore } from 'pages/studyView/StudyViewPageStore';
+import { isUrl, remoteData } from 'cbioportal-frontend-commons';
+import { makeObservable, observable, computed } from 'mobx';
+import { ResourceData, StudyViewFilter } from 'cbioportal-ts-api-client';
+
+export interface IFilesLinksTable {
+    store: StudyViewPageStore;
+}
+
+class FilesLinksTableComponent extends LazyMobXTable<{
+    [id: string]: string | number;
+}> {}
+
+const RECORD_LIMIT = 500;
+
+async function fetchFilesLinksData(
+    filters: StudyViewFilter,
+    sampleIdResourceData: { [sampleId: string]: ResourceData[] },
+    searchTerm: string | undefined,
+    sortAttributeId: string | undefined,
+    sortDirection: 'asc' | 'desc' | undefined,
+    recordLimit: number
+) {
+    const studyClinicalDataResponse = await getAllClinicalDataByStudyViewFilter(
+        filters,
+        searchTerm,
+        sortAttributeId,
+        sortDirection,
+        recordLimit,
+        0
+    );
+    const getResourceDataOfPatients = async () => {
+        const resourcesPerPatient = _(studyClinicalDataResponse.data)
+            .flatMap(clinicaldataItems => clinicaldataItems)
+            .uniqBy('patientId')
+            .map(resource =>
+                internalClient.getAllResourceDataOfPatientInStudyUsingGET({
+                    studyId: resource.studyId,
+                    patientId: resource.patientId,
+                    projection: 'DETAILED',
+                })
+            )
+            .flatten()
+            .value();
+
+        return Promise.all(resourcesPerPatient).then(resourcesPerPatient =>
+            _(resourcesPerPatient)
+                .flatMap()
+                .groupBy('patientId')
+                .value()
+        );
+    };
+
+    const resourcesForPatients = await getResourceDataOfPatients();
+    const buildItemsAndResources = (resourceData: {
```

Comment:
this function might as well be declared outside of fetchFilesLinksData.  i don't think there's a reason to declare it inside here.   in general is a bad practice because it makes it harder to test and in an iterative setting can be very redundant.